rusoto/credential/src/lib.rs

```diff
@@ -100,9 +102,13 @@ impl AwsCredentials {
 
     /// Determine whether or not the credentials are expired.
     fn credentials_are_expired(&self) -> bool {
-        // This is a rough hack to hopefully avoid someone requesting creds then sitting on them
-        // before issuing the request:
-        self.expires_at < Utc::now() + Duration::seconds(20)
+        match self.expires_at {
+            Some(ref e) =>
+                // This is a rough hack to hopefully avoid someone requesting creds then sitting on them
+                // before issuing the request:
+               *e < Utc::now() + Duration::seconds(20),
+            None => false,
```

Comment:
If I'm reading this right, this change modifies behavior so some type of credential providers will *never* refresh. Is that what we want?

I've seen profile provider files changed on the fly and those changes expected to be picked up. But maybe that's not something we want to encourage. 😉

Reply:
If so, then we should rather check the profile file for changes (e.g. using its modification time and/or a checksum) instead of assuming an arbitrary expiration of ten minutes.
But changing the profile file would be a bad practice and I doubt whether any of the other SDKs support that, but I let me check that first.

Reply:
@alex-berger Did you manage to check what other SDK do here?